cranelift/filetests/src/test_run.rs

```diff
@@ -118,20 +132,20 @@ impl SubTest for TestRun {
             return Ok(());
         }
 
-        // We can't use the requested ISA directly since it does not contain info
-        // about the operating system / calling convention / etc..
-        //
-        // Copy the requested ISA flags into the host ISA and use that.
-        let isa = build_host_isa(false, context.flags.clone(), requested_isa.isa_flags());
-
         let test_env = RuntestEnvironment::parse(&context.details.comments[..])?;
 
-        let mut compiler = SingleFunctionCompiler::new(isa);
         for comment in context.details.comments.iter() {
             if let Some(command) = parse_run_command(comment.text, &func.signature)? {
                 trace!("Parsed run command: {}", command);
 
-                let compiled_fn = compiler.compile(func.clone().into_owned())?;
+                // We can't use the requested ISA directly since it does not contain info
+                // about the operating system / calling convention / etc..
+                //
+                // Copy the requested ISA flags into the host ISA and use that.
+                let isa = build_host_isa(false, context.flags.clone(), requested_isa.isa_flags());
+
+                let compiled_fn =
+                    SingleFunctionCompiler::new(isa).compile(func.clone().into_owned())?;
```

Comment:
I'd prefer not to move this code into the loop, if my earlier suggestion on shared references to the `TargetIsa` works out.